src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -2067,4 +2068,16 @@ public static void OutputTrace(string s)
 
         public static string IdOf(object o) => o == null ? "(null)" : $"{o.GetType().Name}#{o.GetHashCode():X2}";
     }
+
+    // struct wrapper is used in order to improve the performance of the epoll thread hot path by up to 3% of some TechEmpower benchmarks
+    // the goal is to have a dedicated generic instantiation and using:
+    // System.Collections.Concurrent.ConcurrentDictionary`2[System.IntPtr,System.Net.Sockets.SocketAsyncContextWrapper]::TryGetValueInternal(!0,int32,!1&)
+    // instead of:
+    // System.Collections.Concurrent.ConcurrentDictionary`2[System.IntPtr,System.__Canon]::TryGetValueInternal(!0,int32,!1&)
```

Comment:
Curious that this would perform better. Why is the dedicated generic instantiation better?

Reply:
Do we ever update the value for an existing key in the dictionary?  If we do, this will make updates more expensive, as they'll be forced to allocate a new node in the CD, whereas with a reference type value, the existing node will be used.

as for why a specific generic instantiation would do better, presumably it's because it's avoiding the generic dictionary lookup, or helping with inlining, or something like that?  Often I see a similar optimization applied as a workaround for removing array covariance checks, but that's on writes, which we shouldn't be doing here frequently.

Reply:
> Do we ever update the value for an existing key in the dictionary?

We don't. We use incremental keys for each AsyncContext. When we run out of keys, we start a new SocketEngine.

> as for why a specific generic instantiation would do better, presumably it's because it's avoiding the generic dictionary lookup, or helping with inlining, or something like that? 

I'm curious what it is.